### ISSUE TITLE
Can't confirm buildings (or anything else?)

### ISSUE THREAD

**Issue by pvh** [Lurkars/gloomhavensecretariat#417]
### Describe the bug

When we try to upgrade buildings at the end of scenarios (for example, just now at the end of Scenario 34) we often can't get Confirm to enable.

### Game Enviroment

We're running 0.78.2, and just finished Scenario 34, retired a character (to reach prosperity level 2), and then could not confirm anything.

### To Reproduce

1. Open the Campaign Sheet
2. Scroll to buildings and try to build a building with an upgrade option available (We chose IIRC the second wall, Wall K)
3. Pay the costs until everything in the left column is paid for.
4. Try pressing Confirm but it is greyed out and does not highlight on mouseover.

### Expected behavior

Clicking confirm would spend the resources and build the building.

### Gloomhaven Secretariat Version

0.78.2

### Used Browser

Chrome

### OS

Windows

### Additional

We reproduced this on several machien and ultimately worked around it by force-building and subsequently paying the costs manually.

I had a look in the code in upgrade-dialog and didn't see anything obviously amiss. I wondered if it was maybe prosperity-based but didn't seem that way. I suspect the button is being inappropriately disabled. Happy to provide more debug info to help track down specifics.

**Comment by pvh** [comment]
Did a little digging. I don't know a whole lot about Angular but this looks like a calculation that isn't updating correctly.
```
> ctx_r7.spent
{gold: 0, hide: 2, lumber: 3, metal: 2, prosperity: 0}gold: 0hide: 2lumber: 3metal: 2prosperity: 0[[Prototype]]: Object
> ctx_r7.paidResources
4
```

**Comment by Lurkars** [comment]
I just forgot to update the `paidResources` variable when initially take the resources from Frosthaven Supply.

Should be fixed in [v0.78.3](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.78.3). Please verify and close issue if solved.

**Comment by pvh** [comment]
Ah, hah! I just found the line myself :) Yep, it works!
